examples/plot_glm_retinal_ganglia.py

```diff
@@ -0,0 +1,259 @@
+# -*- coding: utf-8 -*-
+"""
+================================================================
+GLM for Spike Train Prediction in Primate Retinal Ganglion Cells
+================================================================
+
+* Original tutorial adapted from Johnathan Pillow, Princeton University
+* Dataset provided by E.J. Chichilnisky, Stanford University
+* The dataset is granted by the original authors for educational use only
+* Please contact ``pillow@princeton.edu`` if using beyond its purposes
+
+The original MATLAB and Python tutorial can be found from
+https://github.com/pillowlab/GLMspiketraintutorial.
+
+These data were collected by Valerie Uzzell in the lab of
+E.J. Chichilnisky at the Salk Institute. For full information see
+Uzzell et al. [1]_, or Pillow et al. [2]_.
+
+In this tutorial, we will demonstrate how to fit linear GLM and
+Poisson GLM to predict the spike counts
+recorded from primate retinal ganglion cells.
+The dataset contains spike responses from 2 ON and 2 OFF parasol
+retinal ganglion cells (RGCs) in primate retina, stimulated with
+full-field `binary white noise`. Two experiment performed consisted of a
+long (20-minute) binary stochastic (non-repeating) stimulus
+which can be used for computing the spike-triggered average
+(or characterizing some other model of the response).
+
+References
+----------
+.. [1] Uzzell, V. J., and E. J. Chichilnisky. "Precision of
+   spike trains in primate retinal ganglion cells."
+   Journal of Neurophysiology 92.2 (2004)
+.. [2] Pillow, Jonathan W., et al. "Prediction and decoding of
+   retinal ganglion cell responses with a probabilistic
+   spiking model." Journal of Neuroscience 25.47 (2005)
+
+"""
+
+# Authors: Jonathan Pillow <pillow@princeton.edu>
+#          Titipat Achakulvisut <my.titipat@gmail.com>
+# License: MIT
+
+########################################################
+#
+# Import all the relevance libraries.
+
+import os.path as op
+import json
+
+import numpy as np
+from scipy.linalg import hankel
+
+from pyglmnet import GLM
+
+import matplotlib.pyplot as plt
+from tempfile import TemporaryDirectory
+
+########################################################
+#
+# Fetch the dataset. The JSON file contains the
+# following keys:  ``stim`` (binary stochastic stimulation),
+# ``stim_times`` (time of the stimulation), and
+# ``spike_times`` (recorded time of the spikes)
+
+from pyglmnet.datasets import fetch_rgc_data
+
+with TemporaryDirectory(prefix="tmp_glm-tools") as temp_dir:
+    dpath = fetch_rgc_data(dpath=temp_dir, accept_rgcs_license=True)
```

Comment:
this will mean that users who download our script will not give consent to accepting the license. Unfortunately in Pyglmnet we don't have a very sophisticated download system for datasets. Ideally what you'd want to do is download the dataset in circleCI as a separate command:

```sh
- python -c "import pyglmnet; pyglmnet.datasets.fetch_rgc_data(accept_rgcs_license=True);"
```
And here in the example you would call the function with `accept_rgcs_license=False`. But it will only be invoked for users since for continuous integration, the data is already downloaded.

Reply:
@jasmainak I tried to follow an example from MNE (https://github.com/mne-tools/mne-python/blob/master/mne/datasets/brainstorm/bst_raw.py#L28-L33). However, I'm kinda stuck doing it. Can I ask for your help on this comment?

Reply:
MNE uses a config system. The download path is stored in a config file. It might be too much for one PR to incorporate that as well. We can do that in a subsequent PR. Just make it `False` in the example here and use the same path that you use in `CircleCI`. Maybe `$HOME/glm-data/` ? Not ideal for everyone but we can fix that later.